Microsoft.Alm.Authentication/Secret.cs

```diff
@@ -54,6 +54,22 @@ public static string UriToUrl(TargetUri targetUri, string @namespace)
             return targetName;
         }
 
+        /// <summary>
+        ///     Generate a key based on the ActualUri.
+        ///     This may include username, port, etc
+        /// </summary>
+        public static string UriToActualUri(TargetUri targetUri, string @namespace)
+        {
+            BaseSecureStore.ValidateTargetUri(targetUri);
+            if (String.IsNullOrWhiteSpace(@namespace))
+                throw new ArgumentNullException(@namespace);
+
+            string targetName = $"{@namespace}:{targetUri.ActualUri.AbsoluteUri}";
```

Comment:
As you say the end result is the same, at the moment. However I decided to be as literal as possible, the Bitbucket code uses the ActualUri to check for users/add users so I thought it best to rely on the ActualUri directly.